### PR TITLE
plan: always chose the smaller child as outer for IndexJoin

### PR DESCRIPTION
**Pull request by zz-jason**
## What have you changed? (mandatory)

Always chose the smaller child as the outer one for `IndexJoin` if possible.

## What are the type of the changes (mandatory)?

- Improvement (non-breaking change which is an improvement to an existing feature)

## How has this PR been tested (mandatory)?

- unit test
- explain test

### CODE REVIEW

File: plan/plan.go [hunk 1]
```diff
@@ -320,6 +323,13 @@ type basePlan struct {
 	stats *statsInfo
 }
 
+func (p *basePlan) cardinality() float64 {
+	if p.stats == nil {
+		return math.MaxFloat64
```
Comment (shenli) [inline plan/plan.go]: Why return MaxFloat64 here?

File: plan/plan.go [hunk 2]
```diff
@@ -163,6 +164,8 @@ type LogicalPlan interface {
 	// deriveStats derives statistic info between plans.
 	deriveStats() (*statsInfo, error)
 
+	cardinality() float64
```
Comment (shenli) [inline plan/plan.go]: Add comment for it.

File: plan/gen_physical_plans.go [hunk 3]
```diff
@@ -385,22 +385,30 @@ func (p *LogicalJoin) tryToGetIndexJoin(prop *requiredProp) ([]PhysicalPlan, boo
 			plans = append(plans, join...)
 		}
 	case InnerJoin:
-		join := p.getIndexJoinByOuterIdx(prop, 0)
-		if join != nil {
+		lhsCardinality := p.Children()[0].cardinality()
```
Comment (shenli) [inline plan/gen_physical_plans.go]: Can we move this to line 402?

File: plan/plan.go [hunk 4]
```diff
@@ -320,6 +323,13 @@ type basePlan struct {
 	stats *statsInfo
 }
 
+func (p *basePlan) cardinality() float64 {
+	if p.stats == nil {
+		return math.MaxFloat64
+	}
+	return p.stats.count
```
Comment (lamxTyler) [inline plan/plan.go]: It is not cardinality, cardinality is the number of elements in a set.

File: plan/plan.go (hunk 2 above)
Comment (coocood) [inline plan/plan.go]: Can we add `getStats` instead?
Then use the `count` to determine the outer table.

**Pull request closed without merging**